src/runtime_src/core/edge/tools/xbutil/xbutil.h

```diff
@@ -524,7 +524,47 @@ class device {
         return 0;
     }
 
-    int program(const std::string& xclbin, unsigned region) { std::cout << "Unsupported API " << std::endl; return -1; }
+/* 
+ * program    
+ */
+    int program(const std::string& xclbin, unsigned region) {
+        std::ifstream stream(xclbin.c_str());
+
+        if(!stream.is_open()) {
+            std::cout << "ERROR: Cannot open " << xclbin << ". Check that it exists and is readable." << std::endl;
+            return -ENOENT;
+        }
+
+        if(region) {
+            std::cout << "ERROR: Not support other than -r 0 " << std::endl;
+            return -EINVAL;
+        }
+
+        char temp[8];
+        stream.read(temp, 8);
+
+        if (std::strncmp(temp, "xclbin0", 8)) {
+            if (std::strncmp(temp, "xclbin2", 8))
+                return -EINVAL;
+        }
+
+
+        stream.seekg(0, stream.end);
+        int length = stream.tellg();
+        stream.seekg(0, stream.beg);
+
+        char *buffer = new char[length];
```

Comment:
Unmanaged new / delete are not a good idea.

  std::vector<char> header(length);
  stream.read(header.data(),length);